lib/AppInfo/Application.php

```diff
@@ -172,4 +172,25 @@ function ($mountPoint, $storage) {
 			1
 		);
 	}
+
+	/**
+	 * Probing all modes in a sane order
+	 */
+	public function autoProbe() {
+		$appConfig = $this->getContainer()->query('AppConfig');
+		$scannerFactory = $this->getContainer()->query('ScannerFactory');
+		$set = false;
+		foreach (['daemon', 'socket'] as $mode) {
+			$appConfig->setAvMode($mode);
+			if ($scannerFactory->testConnection($appConfig) === true) {
+				$set = true;
+				break;
+			}
+		}
+
+		if ($set === false) {
+			$appConfig->setAvMode('executable');
+		}
+		\OC::$server->getConfig()->setAppValue('files_antivirus', 'autoprobe', true);
```

Comment:
I think we can get it from the container.

Reply:
updated